src/pages/home/ReportScreen.tsx

```diff
@@ -14,154 +17,119 @@ import OfflineWithFeedback from '@components/OfflineWithFeedback';
+    report: OnyxEntry<ReportWithoutHasDraft>;
 
-            /** The type of the policy */
-            type: PropTypes.string,
-        }),
-    ),
+    /** The report metadata loading states */
+    reportMetadata: OnyxEntry<OnyxTypes.ReportMetadata>;
 
-    /** The account manager report ID */
-    accountManagerReportID: PropTypes.string,
+    /** The report's parentReportAction */
+    parentReportAction: OnyxEntry<OnyxTypes.ReportAction>;
+};
 
+type OnyxHOCProps = {
     /** Onyx function that marks the component ready for hydration */
-    markReadyForHydration: PropTypes.func,
-
-    /** Whether user is leaving the current report */
-    userLeavingStatus: PropTypes.bool,
-
-    viewportOffsetTop: PropTypes.number.isRequired,
-    ...withCurrentReportIDPropTypes,
+    markReadyForHydration?: () => void;
 };
 
-const defaultProps = {
-    isSidebarLoaded: false,
-    reportActions: {},
-    parentReportAction: {},
-    report: {},
-    reportMetadata: {
-        isLoadingInitialReportActions: true,
-        isLoadingOlderReportActions: false,
-        isLoadingNewerReportActions: false,
-    },
-    isComposerFullSize: false,
-    betas: [],
-    policies: {},
-    accountManagerReportID: null,
-    userLeavingStatus: false,
-    markReadyForHydration: null,
-    ...withCurrentReportIDDefaultProps,
-};
+type ReportScreenNavigationProps = StackScreenProps<CentralPaneNavigatorParamList, typeof SCREENS.REPORT>;
+
+type ReportScreenProps = OnyxHOCProps & ViewportOffsetTopProps & CurrentReportIDContextValue & ReportScreenOnyxProps & ReportScreenNavigationProps;
 
-/**
- * Get the currently viewed report ID as number
- *
- * @param {Object} route
- * @param {Object} route.params
- * @param {String} route.params.reportID
- * @returns {String}
- */
-function getReportID(route) {
+/** Get the currently viewed report ID as number */
+function getReportID(route: ReportScreenNavigationProps['route']): string {
     // The report ID is used in an onyx key. If it's an empty string, onyx will return
     // a collection instead of an individual report.
     // We can't use the default value functionality of `lodash.get()` because it only
     // provides a default value on `undefined`, and will return an empty string.
     // Placing the default value outside of `lodash.get()` is intentional.
-    return String(lodashGet(route, 'params.reportID') || 0);
+    return String(route.params?.reportID ?? 0);
 }
 
 function ReportScreen({
-    betas,
+    betas = [],
     route,
     report: reportProp,
-    reportMetadata,
-    reportActions,
+    reportMetadata = {
+        isLoadingInitialReportActions: true,
+        isLoadingOlderReportActions: false,
+        isLoadingNewerReportActions: false,
+    },
+    reportActions = [],
     parentReportAction,
     accountManagerReportID,
     markReadyForHydration,
-    policies,
-    isSidebarLoaded,
+    policies = {},
+    isSidebarLoaded = false,
     viewportOffsetTop,
-    isComposerFullSize,
-    errors,
```

Comment:
`errors` were used in hook dependency only, I wasn't able to find the source of this param as well, so I decided to remove it. If you have any thoughts - let me know